bin/comment_with_dependency_diff

```diff
@@ -0,0 +1,153 @@
+    echo ".gradle files have been changed. Looking for caused dependency changes"
+else
+    echo ".gradle files haven't been changed. There is no need to run the diff"
+    exit 0
+fi
+
+mkdir -p "$DIFF_DEPENDENCIES_FOLDER"
+
+echo "--> Generating dependencies to the file $HEAD_BRANCH_DEPENDENCIES_FILE"
+./gradlew :"$MODULE":dependencies --configuration "$CONFIGURATION" >$HEAD_BRANCH_DEPENDENCIES_FILE
+
+echo "--> Generating build environment dependencies for the head branch"
+./gradlew buildEnvironment >$HEAD_BRANCH_BUILD_ENV_FILE
+
+echo "--> Generating dependencies to the file $BASE_BRANCH_DEPENDENCIES_FILE"
+
+if ! git diff --quiet; then
+  echo "Changes detected. Stashing and continuing with the script"
+  git diff
+  git stash
+fi
+
+git checkout "$baseBranch"
+./gradlew :"$MODULE":dependencies --configuration "$CONFIGURATION" >$BASE_BRANCH_DEPENDENCIES_FILE
+
+echo "--> Generating build environment dependencies for the base branch"
+./gradlew buildEnvironment >$BASE_BRANCH_BUILD_ENV_FILE
+
+git checkout "-"
+
+echo "--> Downloading dependency-tree-diff.jar"
+curl -v -L "https://github.com/JakeWharton/dependency-tree-diff/releases/download/$DEPENDENCY_TREE_VERSION/dependency-tree-diff.jar" -o dependency-tree-diff.jar
+sha=$(sha1sum dependency-tree-diff.jar | cut -d' ' -f1)
+if [[ $sha != "39c50f22cd4211a3e52afec43120b7fdf90f9890" ]]; then
+  echo "dependency-tree-diff.jar file has unexpected sha1"
+  exit 1
+fi
+chmod +x dependency-tree-diff.jar
+
+echo "--> Running dependency-tree-diff.jar"
+./dependency-tree-diff.jar $BASE_BRANCH_DEPENDENCIES_FILE $HEAD_BRANCH_DEPENDENCIES_FILE >$DIFF_DEPENDENCIES_FILE
+
+./dependency-tree-diff.jar $BASE_BRANCH_BUILD_ENV_FILE $HEAD_BRANCH_BUILD_ENV_FILE >$DIFF_BUILD_ENV_FILE
+
+generate_project_dependencies_comment_body() {
+  cat <<EOF
+
+The following changes in project dependencies were detected (configuration \`$CONFIGURATION\`):
+
+<details><summary>expand</summary>
+
+\`\`\`diff
+$(cat "$DIFF_DEPENDENCIES_FILE")
+\`\`\`
+
+</details>
+
+
+EOF
+}
+
+generate_build_dependencies_comment_body() {
+  cat <<EOF
+
+The following changes in the build classpath were detected:
+
+<details><summary>expand</summary>
+
+\`\`\`diff
+$(cat "$DIFF_BUILD_ENV_FILE")
+\`\`\`
+
+</details>
+
+EOF
+}
+
+if [ -n "$DIFF_DEPENDENCIES_FILE" ] || [ -n "$DIFF_BUILD_ENV_FILE" ]; then
+  echo "--> Commenting result to GitHub"
+
+  # Create or clear the comment body file
+  COMMENT_FILE="$DIFF_DEPENDENCIES_FOLDER/comment_body.txt"
+  true > $COMMENT_FILE  # 'true' used to clear the file as a no-op
+
+  append_content() {
+    local content="$1"
+    local github_pr_comment_max_size=65536
+    local current_file_size
+    current_file_size=$(wc -c < "$COMMENT_FILE")
+
+    if (( current_file_size + ${#content} > github_pr_comment_max_size )); then
+      echo "Content exceeds $github_pr_comment_max_size characters. Navigate to Buildkite build artifacts to see details." >> $COMMENT_FILE
```

Comment:
💡  I think we could even add a link to the job for quicker access (users would still need to click on the "Artifacts" tab but at least they'd already be on the right build and job 🙂 )
```suggestion
      echo "Content exceeds $github_pr_comment_max_size characters. [Navigate to Buildkite build artifacts](${BUILDKITE_BUILD_URL}#${BUILDKITE_JOB_ID}) to see details." >> $COMMENT_FILE
```

Reply:
Btw:
 - There's a risk that `current_file_size` would be so close to `github_pr_comment_max_size` already that even just adding this "Content exceed …" line to it will make it go over the limit…
    - So the `github_pr_comment_max_size` limit value should really be 65536… minus the length of this line telling that the content exceeds the max size 😅 
 - Wouldn't the logic of `append_content` here add the `echo "Content exceeds …"` string multiple times if you call `append_content` multiple times and it's not just the last one but one before that that makes the length be exceeded?
    - Though I guess in this case it's not a big deal because you only call `append_content` a maximum number of 2 times, so this scenario can only happen if the content from `DIFF_DEPENDENCIES_FILE` is already too long, and in which case you'll print the `Content exceeds …` 2 times (one for when you call it for `DIFF_DEPENDENCIES_FILE`, then other when you call it for `DIFF_BUILD_ENV_FILE`)… so probably not that dramatic and not worth fixing I guess.
    - But just wanted to mention in case you wanted to look into it, or if you planned in future PRs to potentially add more info and call `append_content` more times in future iterations of this helper, just to keep that in mind.